sympy/matrices/matrices.py

```diff
@@ -4278,59 +4278,239 @@ def entry_U(i, j):
+            \end{bmatrix}
+
+        .. math::
+            U = \begin{bmatrix}
+            U_{0, 0} & U_{0, 1} & U_{0, 2} & \cdots & U_{0, n-1} \\
+            0 & U_{1, 1} & U_{1, 2} & \cdots & U_{1, n-1} \\
+            0 & 0 & U_{2, 2} & \cdots & U_{2, n-1} \\
+            \vdots & \vdots & \vdots & \ddots & \vdots \\
+            0 & 0 & 0 & \cdots & U_{n-1, n-1}
+            \end{bmatrix}
+
+        And for a matrix with more rows than the columns,
+        the decomposition would look like:
+
+        .. math::
+            L = \begin{bmatrix}
+            1 & 0 & 0 & \cdots & 0 & 0 & \cdots & 0 \\
+            L_{1, 0} & 1 & 0 & \cdots & 0 & 0 & \cdots & 0 \\
+            L_{2, 0} & L_{2, 1} & 1 & \cdots & 0 & 0 & \cdots & 0 \\
+            \vdots & \vdots & \vdots & \ddots & \vdots & \vdots & \ddots
+            & \vdots \\
+            L_{n-1, 0} & L_{n-1, 1} & L_{n-1, 2} & \cdots & 1 & 0
+            & \cdots & 0 \\
+            L_{n, 0} & L_{n, 1} & L_{n, 2} & \cdots & L_{n, n-1} & 1
+            & \cdots & 0 \\
+            \vdots & \vdots & \vdots & \ddots & \vdots & \vdots
+            & \ddots & \vdots \\
+            L_{m-1, 0} & L_{m-1, 1} & L_{m-1, 2} & \cdots & L_{m-1, n-1}
+            & 0 & \cdots & 1 \\
+            \end{bmatrix}
+
+        .. math::
+            U = \begin{bmatrix}
+            U_{0, 0} & U_{0, 1} & U_{0, 2} & \cdots & U_{0, n-1} \\
+            0 & U_{1, 1} & U_{1, 2} & \cdots & U_{1, n-1} \\
+            0 & 0 & U_{2, 2} & \cdots & U_{2, n-1} \\
+            \vdots & \vdots & \vdots & \ddots & \vdots \\
+            0 & 0 & 0 & \cdots & U_{n-1, n-1} \\
+            0 & 0 & 0 & \cdots & 0 \\
+            \vdots & \vdots & \vdots & \ddots & \vdots \\
+            0 & 0 & 0 & \cdots & 0
+            \end{bmatrix}
+
+        Finally, for a matrix with more columns than the rows, the
+        decomposition would look like:
+
+        .. math::
+            L = \begin{bmatrix}
+            1 & 0 & 0 & \cdots & 0 \\
+            L_{1, 0} & 1 & 0 & \cdots & 0 \\
+            L_{2, 0} & L_{2, 1} & 1 & \cdots & 0 \\
+            \vdots & \vdots & \vdots & \ddots & \vdots \\
+            L_{m-1, 0} & L_{m-1, 1} & L_{m-1, 2} & \cdots & 1
+            \end{bmatrix}
+
+        .. math::
+            U = \begin{bmatrix}
+            U_{0, 0} & U_{0, 1} & U_{0, 2} & \cdots & U_{0, m-1}
+            & \cdots & U_{0, n-1} \\
+            0 & U_{1, 1} & U_{1, 2} & \cdots & U_{1, m-1}
+            & \cdots & U_{1, n-1} \\
+            0 & 0 & U_{2, 2} & \cdots & U_{2, m-1}
+            & \cdots & U_{2, n-1} \\
+            \vdots & \vdots & \vdots & \ddots & \vdots
+            & \cdots & \vdots \\
+            0 & 0 & 0 & \cdots & U_{m-1, m-1}
+            & \cdots & U_{m-1, n-1} \\
+            \end{bmatrix}
+
+        About the compressed LU storage:
+
+        The results of the decomposition are often stored in compressed
+        forms rather than returning $L$ and $U$ matrices individually.
```

Comment:
Is this also the case in other CAS?

Knowing what I do about memory consumption and management in Python/sympy this is not really saving anything. The matrices are inherently small and the matrix object itself consumes less memory than the expressions it references.

Reply:
This format would be used in [Mathematica](https://reference.wolfram.com/language/ref/LUDecomposition.html).

I agree that it's not having much useful for CAS like SymPy, than wolfram who is also dealing with fast numerics.
But 'storage efficiency' is the only clue that I can find from some books that does exactly same like this.